### ISSUE TITLE
Question: is rounding necessary for proportions?

### ISSUE THREAD

**Issue by zkamvar** [R4EPI/sitrep#40]
I noticed that all of the proportion related functions (in cfr.R) have the `digits` argument, which defaults to 1. This causes rounding to a single decimal place for the results. Would a better solution just be to modify the "digits" option?

``` r
(x <- tibble::enframe(runif(5)))
#> # A tibble: 5 x 2
#>    name value
#>   <int> <dbl>
#> 1     1 0.981
#> 2     2 0.250
#> 3     3 0.279
#> 4     4 0.303
#> 5     5 0.276
knitr::kable(x)
```

<table>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.9806477</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.2499407</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.2786514</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.3034308</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.2764433</td>
</tr>
</tbody>
</table>

``` r
options(digits = 3)
knitr::kable(x)
```

<table>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.981</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.250</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.279</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.303</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.276</td>
</tr>
</tbody>
</table>

``` r
options(digits = 7)
knitr::kable(x, digits = 3)
```

<table>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.981</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.250</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.279</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.303</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.276</td>
</tr>
</tbody>
</table>

<sup>Created on 2019-01-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by zkamvar** [comment]
I just found out that the digits option is different from the digits
specification in knitr kables

``` r
x <- tibble::enframe(runif(3) + 69)
knitr::kable(x, caption = "default: prints 7 digits")
```

<table>
<caption>default: prints 7 digits</caption>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">69.71135</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">69.97812</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">69.25433</td>
</tr>
</tbody>
</table>

``` r
options(digits = 2)
knitr::kable(x, caption = "options(digits = 2)")
```

<table>
<caption>options(digits = 2)</caption>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">69</td>
</tr>
</tbody>
</table>

``` r
options(digits = 7)
knitr::kable(x, digits = 2, caption = "options(digits = 7); kable(x, digits = 2)")
```

<table>
<caption>options(digits = 7); kable(x, digits = 2)</caption>
<thead>
<tr class="header">
<th style="text-align: right;">name</th>
<th style="text-align: right;">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">69.71</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">69.98</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">69.25</td>
</tr>
</tbody>
</table>

<sup>Created on 2019-01-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by aspina7** [comment]
think its simpler to leave within the function - people can then also use the funciton outside of an rmd if wanted.... just my reasoning but happen to be corrected!

**Comment by dirkschumacher** [comment]
I am a bit indifferent, not sure what is easier for folks.

**Comment by zkamvar** [comment]
> think its simpler to leave within the function - people can then also use the funciton outside of an rmd if wanted.... just my reasoning but happy to be corrected!

I see these functions as helpers for calculating a certain value. These can be used for displaying tables, but can also be used for plotting and direct comparison. In the Rmarkdown document, we display the tables through `kable()`, which already has its own rounding facilities.

Besides, since the function returns a tibble, the rounding is not necessary even for those who don't use `kable()` since tibbles only print the significant digits by default. 

> Also more explaining to do if have to specify in options and in kable the num digits wanted...

I messed up the explanation here. The only thing that would be done is to move the digits argument to `kable()`, no messing around with options.

**Comment by aspina7** [comment]
cool beans

**Comment by zkamvar** [comment]
closed by #42
